xbmc/windowing/X11/GLContextEGL.h

```diff
@@ -23,6 +23,7 @@
 #if defined(HAVE_X11)
 #include "GLContext.h"
 #include "EGL/egl.h"
+#include <X11/Xutil.h>
```

Comment:
that is because kodi for rockchip linux is just an X11 application that uses OpenGLES to render it graphics into its window. so it isnt really a platform actually.

however it seems that with the graphical GPU driver of the rockchip SoC that implements OpenGLES (ARM mali GPU) has a different egl header file than usual for kodi (probably mesa based). 

so the include files in mali egl package (/usr/lib/arm-linux-gnueabihf/pkgconfig/egl.pc) dont include any <X11/Xutil.h> inside. but for instance the include files in the mesa egl package on ordinary pc do (/lib/x86_64-linux-gnu/pkgconfig/egl.pc) .

this implicit inclusion of XUtil.h in <egl/platform.h> is used in GLContextEGL.cpp to define the XVisualInfo type that is used in its X11 calls. this doesnt work for mali egl package and hence Xutil.h needs to be included explicitly.

hope things are a bit clearer now?